### PR TITLE
Introduction Chapter Finished / CLA Assigned

### PR DESCRIPTION
**Pull request by KoreaComK**
Finished the introduction chapter.
I didn't changed the chapters files names since @nyxnor said that he started to make this changes.

### CODE REVIEW

File: 01_0_Introduction.md [hunk 1]
```diff
@@ -1,63 +1,64 @@
-# Chapter One: Introduction to Learning Bitcoin Core (& Lightning) from the Command Line
 
-## Introduction
+# CAPÍTULO 1: Introdução à Aprendizagem do Bitcoin Core (& Lightning Network) na linha de comando
```
Comment (lukedevj) [inline 01_0_Introduction.md:2]: <p> Instead of using "na linha de comando" use "pela linha de comando" </p>

File: 01_1_Introducing_Bitcoin.md [hunk 2]
```diff
@@ -1,148 +1,150 @@
...
+## Sobre o Bitcoin
 
-Obviously, Bitcoin is the heart of this book, but it's also the originator of many other systems, including blockchains and Lightning, which are both detailed in this tutorial, and many other cryptocurrencies such as Ethereum and Litecoin, which are not.
+O Bitcoin é um sistema programável que permite a transferência da moeda bitcoin. Tem como base um sistema descentralizado _peer-to-peer_ (de ponta à ponta) de nodes, que inclui full nodes, carteiras e mineradores. Trabalhando juntos, eles garantem que as transações do bitcoin sejam rápidas e irreversíveis. Graças à natureza descentralizada do sistema, essas transações também são resistentes à censura e podem fornecer outras vantagens, como pseudonimato, caso sejam bem utilizadas.
```
Comment (lukedevj) [inline 01_1_Introducing_Bitcoin.md:8]: "pseudonimato" to "pseudo-anonimato".

Comment (lukedevj) [inline 01_1_Introducing_Bitcoin.md:8]: "pseudonimato" to "pseudo-anonimato".

**Review by lukedevj — CHANGES_REQUESTED** [review]

**Comment by KoreaComK** [comment]
All changes included in the PR.

File: pt/15_1_Accessing_Bitcoind_with_C.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
+
+# 15.1: Acessando o Bitcoind usando C com bibliotecas RPC
+
+> :information_source: **NOTA:** Esta seção foi adicionada recentemente ao curso e é um esboço que ainda pode estar aguardando revisão. Portanto, leitor, tenha cuidado.
+
+Você já viu uma maneira alternativa de acessar as portas RPC do Bitcoind: Usando o ``curl``, que cobrimos no [Capítulo 4 Prefácio](04_4__interlude_using_curl.md). Interagir com o ``Bitcoind`` através de uma biblioteca de RPC usando C não é diferente do que já vimos, só precisamos de boas bibliotecas para nos auxiliar. Esta seção introduz um pacote chamado ``libbitcoinrpc``, que permite acessar a porta JSON-RPC do ``bitcoind``. Ele usa uma biblioteca ``curl`` para acessar os dados e usa a biblioteca ``jansson`` para codificar e decodificar o JSON.
+
+## Configurando o libbitcoinrpc
+
+Para usar o ``libbitcoinrpc``,  precisemos instalar uma configuração básica C e os pacotes dependentes, que são ``libcurl``, ``libjansson``, e ``libuuid``. Depois faremos isso no seu servidor Standup Bitcoin (ou em qualquer outro servidor Ubuntu).
```
Comment (lukedevj) [inline pt/15_1_Accessing_Bitcoind_with_C.md:10]: "precisemos" to "precisaremos"

File: pt/15_1_Accessing_Bitcoind_with_C.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+Suggested packages:
+  libcurl4-doc libidn11-dev libkrb5-dev libldap2-dev librtmp-dev libssh2-1-dev
+The following NEW packages will be installed:
+  libcurl4-openssl-dev libjansson-dev uuid-dev
+0 upgraded, 3 newly installed, 0 to remove and 4 not upgraded.
+Need to get 358 kB of archives.
+After this operation, 1.696 kB of additional disk space will be used.
+Do you want to continue? [Y/n] y
+```
+You can then download [libbitcoinrpc from Github](https://github.com/gitmarek/libbitcoinrpc/blob/master/README.md). Clone it or grab a zip file, as you prefer.
```
Comment (lukedevj) [inline pt/15_1_Accessing_Bitcoind_with_C.md:23]: Remova esta linha.

File: pt/15_1_Accessing_Bitcoind_with_C.md [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+gcc -fPIC -O3 -g -Wall -Werror -Wextra -std=c99 -D VERSION=\"0.2\" -shared -Wl,-soname,libbitcoinrpc.so.0 \
+src/bitcoinrpc_err.o src/bitcoinrpc_global.o src/bitcoinrpc.o src/bitcoinrpc_resp.o src/bitcoinrpc_cl.o src/bitcoinrpc_method.o \
+-o .lib/libbitcoinrpc.so.0.2 \
+-Wl,--copy-dt-needed-entries -luuid -ljansson -lcurl
+ldconfig -v -n .lib
+.lib:
+  libbitcoinrpc.so.0 -> libbitcoinrpc.so.0.2 (changed)
+ln -fs libbitcoinrpc.so.0 .lib/libbitcoinrpc.so
+```
+If that works, you can install the package:
```
Comment (lukedevj) [inline pt/15_1_Accessing_Bitcoind_with_C.md:77]: Esqueceu de traduzir esta parte.

File: pt/15_1_Accessing_Bitcoind_with_C.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+Normalmente definiríamos os parâmetros em seguida, mas o ``GetMiningInfo`` não requer parâmetros, por isso podemos pular essa parte.
+
+Também devemos criar outros dois objetos, um "objeto de resposta" e um "objeto de erro". Eles podem ser inicializados da seguinte forma:
+```
+bitcoinrpc_resp_t *btcresponse  = NULL;
+btcresponse = bitcoinrpc_resp_init();
+
+bitcoinrpc_err_t btcerror;
+```
+You use the `rpc_client` variable that you already learned about in the previous test, and add on your `getmininginfo` method and the two other objects:
```
Comment (lukedevj) [inline pt/15_1_Accessing_Bitcoind_with_C.md:187]: **Remova esta linha**

File: pt/15_1_Accessing_Bitcoind_with_C.md [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+json_t *jsonresponse = NULL;
+jsonresponse = bitcoinrpc_resp_get(btcresponse);
+```
+Se quisermos gerar os resultados completos da chamada RPC no JSON, podemos fazer com uma simples invocação do ``json_dumps``, da biblioteca ``jansson``:
+```
+printf ("%s\n", json_dumps(j, JSON_INDENT(2)));
+```
+No entanto, como agora estamos escrevendo programas completos, provavelmente iremos querer fazer um trabalho mais sutil, como retirar valores individuais do JSON para algum uso específico. A [Referência do Jansson](https6//jansson.readthedocs.Io/en/2.10/apiref.html) traz detalhes de como fazer.
+
+Assim como quando estávamos usando o [curl](04_4__interlude_using_curl.md), descobrimos que o RPC retorna um objeto JSON contendo um ``ID``, um ``error`` e, mais importante, um objeto JSON do tipo ``result``.
```
Comment (lukedevj) [inline pt/15_1_Accessing_Bitcoind_with_C.md:206]: **Assim como ("quando" -> remove) estávamos usando o**

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 8]
```diff
@@ -0,0 +1,355 @@
+
+# 15.2: Programando o Bitcoind usando C com bibliotecas RPC
+
+> :information_source: **NOTE:** This section has been recently added to the course and is an early draft that may still be awaiting review. Caveat reader.
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md:4]: Remove this line.

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+```
+
+> :warning: **ATENÇÃO:** Um programa real calcularia uma taxa que minimizasse o custo, garantindo que a velocidade fosse aquela que o remetente estivesse disposto a utilizar.
+
+### Etapa 3: Preparando nosso RPC
+
+Obviamente, precisaremos preparar todas as nossas variáveis novamente, conforme discutido na sessão [§15.1: Acessando o Bitcoind usando C](15_1_Accessing_Bitcoind_with_C.md). Também precisaremos inicializar a nossa biblioteca, conectar o cliente RPC e preparar nosso objeto de resposta:
+```
+bitcoinrpc_global_init();
+rpc_client = bitcoinrpc_cl_init_params ("bitcoinrpc", "YOUR-RPC-PASSWD", "127.0.0.1", 18332);
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md:74]: "bitcoinrpc_cl_init_params (" -> remove this space before the "("

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+```
+
+No entanto, o verdadeiro trabalho consiste em decodificar a resposta. Na seção anterior vimos que a biblioteca ``jansson`` era "um tanto quanto desajeitada" e esta é a razão: Precisamos criar (e limpar) um conjunto muito grande de objetos ``json_t`` para descobrir o que queremos.
+
+Primeiro, precisamos nos recuperar o campo ``result`` do JSON:
+```
+json_t *lu_response = NULL;
+json_t *lu_result = NULL;
+
+lu_response = bitcoinrpc_resp_get (btcresponse);
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md:93]: remove this space before the (

**Comment by KoreaComK** [comment]
Changes made.
Also including the language specification in the code.

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+if (!tx_id) {
+
+  printf("Very Sad: You don't have any UTXOs larger than %f\n",tx_total);
+  exit(-1);
+}
+```
+
+> **ATENÇÃO** Um programa em produção usaria sub-rotinas para este tipo de pesquisa, de forma que pudéssemos chamar vários RPCs de uma biblioteca de funções C. Vamos apenas colocar tudo em um `main` como parte da nossa filosofia KISS.
+
+### Etapa 5: Criando uma alteração de endereço
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md:160]: Your mistranslation is not a "alteração", it is "Criando um endereço de troco".

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+
+Repita a metodologia padrão de pesquisa RPC para obter uma alteração de endereço:
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md:162]: Same problem as above comment.

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 13]
```diff
@@ -0,0 +1,355 @@
...
+
+if (btcerror.code != BITCOINRPCE_OK) {
+
+printf("Error: listunspent error code %d [%s]\n", btcerror.code,btcerror.msg);
+
+  exit(-1);
+
+}
+
+lu_response = bitcoinrpc_resp_get (btcresponse);
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md]: remove the space before the (

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
+```
+A única diferença é quais informações específicas são extraídas do objeto JSON.
+
+> :warning: **ATENÇÃO:** Aqui temos uma sub-rotina que seria bem legal: Abstrair toda a inicialização e chamada do método RPC.
+
+### Etapa 6: Criando uma transação bruta
+
+Criar a transação bruta real é outra parte complicada da programação da substituição do ``sendtoaddress``. Isso porque requer a criação de um objeto JSON complexo como parâmetro.
+
+Para criarmos esses parâmetros corretamente, precisaremos revisar o que o RPC ``createrawtransaction`` espera que passemos de informação. Felizmente, isso é fácil de determinar usando a funcionalidade ``bitcoin-cli help``:
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md:195]: "espera que passemos de informação" to "espera que passemos como argumento "

File: pt/15_2_Programming_Bitcoind_with_C.md [hunk 15]
```diff
@@ -0,0 +1,355 @@
...
+```
+json_t *params = NULL;
+params = json_array();
+json_array_append(params,inputparams);
+json_array_append(params,outputparams);
+```
+
+#### Etapa 6.4: Fazendo a chamada ao RPC
+
+Vamos usar o método normal para criar da chamada ao RPC:
```
Comment (lukedevj) [inline pt/15_2_Programming_Bitcoind_with_C.md:253]: "Vamos usar o método normal para criar" [da to uma] "chamada ao RPC".

**Review by lukedevj — COMMENTED** [review]
Revised Chapter 15 waiting for change.

**Comment by shannona** [comment]
Hey Folks,

Great to see this coming along, and it looks like you're following a great process with one author, one editor, and making revisions based on that.

Thank you! I'm very much looking forward to this going forward and us having a great Portuguese translation of Learning Bitcoin in the near future.

Two issues that I see right now:

1.) Something in your process is leading you to delete the original English files. Take a look at the diffs, and you'll see the English version of the files you've been translating have been deleted: https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/pull/224/files

Obviously, we need to get that resolved before we can do any merges. My suggestion is that you rebranch and then copy in your Portuguese translations, but whatever works for you is fine, as long as our original files will still be there when we merge.

2.) I'm not getting a good milestone to merge on because by the time one section is revised, the next is being written. Two possible solutions for this:

A.) Pause and give me a milestone to merge in the days I'm around (see below).

B.) Create a new fork for each chapter. This is actually what we'll be suggesting when I revise the TRANSLATING docs, now that we've seen how things work. This way, you create a fork, you add a chapter, you send a PR, you review it, you revise it, and then you alert me that it's finalized and ready for a merge.

Whichever works for you is fine, just under the current methodology this might not get merged until the end, which is subpar.

==
As a general note, I usually spend Tuesday & Wednesday (Hawaiian time) on Blockchain Commons projects, with my other days spent writing for other people (and working on my own books in process). That means that I'll usually delay any comments, merges, etc., until Tuesday, and that if you want to use the first methodology, where you pause and ask me to merge after everything is revised, that'd usually happen 7.30 am - 3.30 pm HST on Tuesday and Wednesday.

Thanks again! Let me know when the issue with the deleted files is resolved (either here or by creating a new PR from a new branch) and when everything is at a point where it's been reviewed and revised and can be merged (by either of the methods i suggest).

**Comment by KoreaComK** [comment]
Changes pointed by @lukedevj of Chapter 15 made.

@shannona thanks for the feedback and for showing a plan to translation.
I already finished to made the changes and also I included the deleted files. So, I think now it could be merged.
I will fork and make a PR one chapter per time from now to future, to get the things more organized.

Cheers!

**Comment by nyxnor** [comment]
@KoreaComK 
> ## Create a new fork for each chapter.
https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/pull/224#issuecomment-866224009

A new fork in a new PR. Keep 1 chapeter per PR.

**Pull request closed without merging**